gridstatus/pjm.py

```diff
@@ -257,29 +267,107 @@ def get_lmp(
+                )
+
         if date == "latest":
-            """Currently only supports DAY_AHEAD_HOURlY"""
+            """Supports DAY_AHEAD_HOURLY, REAL_TIME_5_MIN"""
             return self._latest_lmp_from_today(
                 market=market,
                 locations=locations,
                 location_type=location_type,
                 verbose=verbose,
             )
 
-        if locations == "hubs":
-            locations = self.hub_node_ids
+        recent_threshold = pd.Timestamp.now(tz=self.default_timezone) - pd.Timedelta(
+            days=DV_LMP_RECENT_NUM_DAYS,
+        )
+        dv_df = None
+        if utils._handle_date(date, tz=self.default_timezone) >= recent_threshold:
+            dv_df = self._get_lmp_via_dv(
+                date,
+                market,
+                end=end,
+                locations=locations,
+                location_type=location_type,
+                verbose=verbose,
+            )
+
+        # Figure out whether we need to call the JSON API
+        if utils.is_today(date, tz=self.default_timezone):
+            fetch_json = market in (Markets.DAY_AHEAD_HOURLY,)
+        else:
+            fetch_json = True
+
+        json_df = None
+        if fetch_json:
+            json_df = self._get_lmp_via_pjm_json(
+                date,
+                market,
+                end=end,
+                locations=locations,
+                location_type=location_type,
+                verbose=verbose,
+            )
+
+        dfs = [dv_df, json_df]
+        dfs = [df for df in dfs if df is not None]
 
+        # deduplicate data, choosing the _src=json when there are duplicates
+        # in (time, market, location name)
+        df = self._df_deduplicate(
+            dfs,
+            unique_cols=["Time", "Market", "Location Name"],
+            keep_field="_src",
+            keep_value="json",
+            verbose=verbose,
+        )
+        df.sort_values("Time", inplace=True)
+        df.reset_index(drop=True, inplace=True)
+        df = df[
+            [
+                "Time",
+                "Market",
+                "Location",
+                "Location Name",
+                "Location Type",
+                "LMP",
+                "Energy",
+                "Congestion",
+                "Loss",
+                # "_src",  # uncomment to debug
+            ]
+        ]
+        return df
+
+    def _get_lmp_via_pjm_json(
+        self,
+        date,
+        market: str,
+        end=None,
+        locations=None,
+        location_type=None,
+        verbose=False,
+    ):
+        """Fetches data using the pjm_json api"""
         params = {}
 
         if market == Markets.REAL_TIME_5_MIN:
             market_endpoint = "rt_fivemin_hrl_lmps"
             market_type = "rt"
         elif market == Markets.REAL_TIME_HOURLY:
-            # todo implemlement location type filter
+            # todo implement location type filter
```

Comment:
i think we can get rid of these todos. the location type filter is happening below